sympy/core/tests/test_numbers.py

```diff
@@ -967,6 +967,7 @@ def test_powers():
 def test_integer_nthroot_overflow():
     assert integer_nthroot(10**(50*50), 50) == (10**50, True)
     assert integer_nthroot(10**100000, 10000) == (10**10, True)
+    assert integer_nthroot(100, 10**1000) == (1, False)
```

Comment:
```
In [1]: from sympy.abc import a

In [2]: b = a**a

In [3]: c = b**b

In [4]: c.subs(a, -4)
Out[4]: 2**(31/32)/2

In [5]: c.subs(a, -5)
Out[5]: -(-1)**(3124/3125)*5**(1/625)

In [6]: c.subs(a, -6)
Out[6]: 6**(7775/7776)/6

In [7]: c.subs(a, -7)
Out[7]: -(-1)**(823542/823543)*7**(1/117649)
```
Looks good.
`c.subs(a, -10)` has high usage of system resources but overflow is no longer an issue.

Reply:
Yes, this PR correctly fixes the OverFlow issue. You can try the following example too:

from sympy.abc import a


if __name__ == '__main__':
    b = a ** a
    c = b ** b
    c.subs({a: -144})

Reply:
Yes, this PR correctly fixes the OverFlow issue. You can try the following example too:
```
In [3]: if __name__ == '__main__':
   ...:     b = a ** a
   ...:     c = b ** b
   ...:     c.subs({a: -144})
```